struct.c

```diff
@@ -510,65 +509,123 @@ rb_struct_define_under(VALUE outer, const char *name, ...)
- *  .new takes keyword arguments instead of normal arguments.
- *
- *     Customer = Struct.new(:name, :address, keyword_init: true)
- *     Customer.new(name: "Dave", address: "123 Main")
- *     #=> #<struct Customer name="Dave", address="123 Main">
- *
- *  If a block is given it will be evaluated in the context of
- *  +StructClass+, passing the created class as a parameter:
+ *    Struct.new(class_name) -> new_class
+ *    Struct.new(*member_names, **kwargs) -> new_class
+ *    Struct.new(class_name, *member_names, **kwargs) -> new_class
+ *    Struct.new(class_name) {|new_class| ... } -> new_class
+ *    Struct.new(*member_names, **kwargs) {|new_class| ... } -> new_class
+ *    Struct.new(class_name, *member_names, **kwargs) {|new_class| ... } -> new_class
+ *
+ *  Returns a new subclass of +Struct+.  The new subclass:
+ *
+ *  - May be anonymous, or may have the name given by +class_name+.
+ *  - May have members as given by +member_names+.
+ *  - May have initialization via ordinary arguments (the default)
+ *    or via keyword arguments (if <tt>keyword_init: true</tt> is given).
+ *
+ *  <b>\Class Name</b>
+ *
+ *  With string argument +class_name+,
+ *  returns a new subclass of +Struct+ named <tt>Struct::<em>class_name</em></tt>:
+ *
+ *    # Class name given, but no member names.
+ *    Foo = Struct.new('Foo')             # => Struct::Foo
+ *    Foo.name                            # => "Struct::Foo"
+ *    Foo.superclass                      # => Struct
+ *    # Class name and member names given.
+ *    Foo = Struct.new('Foo', :foo, :bar) # => Struct::Foo
+ *    Foo.name                            # => "Struct::Foo"
+ *    Foo.superclass                      # => Struct
+ *
+ *  Without string argument +class_name+,
+ *  returns a new anonymous subclass of +Struct+:
+ *
+ *    Struct.new(:foo, :bar).name # => nil
+ *
+ *  <b>Member Names</b>
+ *
+ *  With symbol arguments +member_names+,
+ *  returns a new subclass of +Struct+ with the given members:
+ *
+ *    Struct.new(:foo, :bar).members        # => [:foo, :bar]
+ *    Struct.new('Foo', :foo, :bar).members # => [:foo, :bar]
+ *
+ *  The new subclass has instance methods corresponding to +member_names+:
+ *
+ *    Foo = Struct.new('Foo', :foo, :bar)
+ *    Foo.instance_methods(false) # => [:foo, :bar, :foo=, :bar=]
+ *    f = Foo.new                 # => #<struct Struct::Foo foo=nil, bar=nil>
+ *    f.foo                       # => nil
+ *    f.foo = 0                   # => 0
+ *    f.bar                       # => nil
+ *    f.bar = 1                   # => 1
+ *    f                           # => #<struct Struct::Foo foo=0, bar=1>
+ *
+ *  <b><tt>Singleton Methods</tt></b>
+ *
+ *  The new subclass also has these singleton methods:
+ *
+ *    Foo = Struct.new('Foo', :foo, :bar)
+ *    Foo.singleton_methods # => [:[], :new, :inspect, :members]
+ *
+ *  - \Method <tt>:[]</tt> creates an instance of the subclass:
```

Comment:
Since `[]` is an alias to `new`, I would not describe it separately or provide examples of use.

Reply:
Done.